warp10/src/main/java/io/warp10/script/binary/ComparisonOperation.java

```diff
@@ -0,0 +1,215 @@
+//
+//   Copyright 2019  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.binary;
+
+import io.warp10.continuum.gts.GTSHelper;
+import io.warp10.continuum.gts.GTSOpsHelper;
+import io.warp10.continuum.gts.GeoTimeSerie;
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+
+public abstract class ComparisonOperation extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+  
+  public abstract boolean operator(int op1, int op2);
+  
+  //default behavior for > < operators. For >= <= == or !=, you need to set these.
+  public boolean ifOneNaNOperand = false;
```

Comment:
Does it need to be public?

Shouldn't it be final?

Reply:
It needs to be set by subclasses, such NE or EQ or GE or LE. It could be protected instead of public.

Reply:
Fixed with a different constructor.